opm/simulators/linalg/MILU.hpp

```diff
@@ -43,8 +45,38 @@ enum class MILU_VARIANT{
 
 MILU_VARIANT convertString2Milu(const std::string& milu);
 
-namespace detail
+
+  namespace detail 
 {
+  /*  class AuxFunctors
+{
+public:
+  static std::function<double(const double&)> identityFunctor(); 
+  static std::function<double(const double&)> oneFunctor(); 
+  static std::function<double(const double&)> signFunctor(); 
+  static std::function<double(const double&)> isPositiveFunctor(); 
+  static std::function<double(const double&)> absFunctor(); 
+  };*/
+  
+  template <typename T>
+  //T identityFunctor(const T&);
+  std::function<T(const T&)> identityFunctor();
+
+  template <typename T>
+  //T oneFunctor(const T&);
+  std::function<T(const T&)> oneFunctor();
+
+  template <typename T>
+  //T signFunctor(const T&);
+  std::function<T(const T&)> signFunctor();
+
+  template <typename T>
+  //T isPositiveFunctor(const T&);
+  std::function<T(const T&)> isPositiveFunctor();
+
+  template <typename T>
+  // T absFunctor(const T&);
+  std::function<T(const T&)> absFunctor();
```

Comment:
Not sure why you changed that. Seems like the previous version was perfectly ok.
Those functions take a double parameter and return a double. With this change you are now saying that they take no parameter and return a std::function.

Reply:
Somehow I got confused about the` std::function<type1(type2)>` in the front. Thinking that I'm defining/declaring a function that takes type2 objects and returns type1 objects. But there, I'm saying that `absFunctor()` returns a `std::function<type1(type2)>` object. Now, I see the difference/error. Thanks!